### DISCUSSION TITLE
Building multi-stage Docker images with environment variables

### DISCUSSION THREAD

**Discussion by davidkeaveny** [aws/aws-lambda-dotnet#1777] [Q&A] [answered]
Up until now, I have been building lambda functions based on .NET 8 and deploying via zip files. My lambda function consumes NuGet packages from a private NuGet feed, and it's easy enough to configure the builds for that. 

I now would like to try using Docker instead of zip for the lambda functions; I've been able to use the private NuGet feed for other Docker images, I just need to inject environment variables for the username and password to use with the feed; however, I can't see anything in the documentation for `dotnet lambda package` or its cousins on how to pass in environment variables to the Docker build that is being invoked. Is this possible?

If I were to build my lambda in a GitHub Action and wanted to push to ECR, what would that look like in terms of the commands that I need to execute? All the documentation I can find at the moment seems to assume you're deploying from the desktop, which is fine for demo apps, not so good for real-world applications where you have tighter security requirements...

**Comment by bhoradc** [comment, marked as answer]
Hi @davidkeaveny, 

Thank you for submitting this Discussion. We have a Feature Request [#329](https://github.com/aws/aws-extensions-for-dotnet-cli/issues/329) in backlog to `Add options for pre-commands and environment variables to lambda package command`. When this gets implemented, you should be able to build lambda directly in Github Action. 

Regards, 
Chaitanya
